File: Difficulty: Easy/N meetings in one room/n-meetings-in-one-room.java
//{ Driver Code Starts
import java.io.*;
import java.lang.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());

        while (t-- > 0) {
            String inputLine[] = br.readLine().trim().split(" ");
            int n = Integer.parseInt(inputLine[0]);

            int start[] = new int[n];
            int end[] = new int[n];

            inputLine = br.readLine().trim().split(" ");
            for (int i = 0; i < n; i++) start[i] = Integer.parseInt(inputLine[i]);

            inputLine = br.readLine().trim().split(" ");
            for (int i = 0; i < n; i++) end[i] = Integer.parseInt(inputLine[i]);

            int ans = new Solution().maxMeetings(n, start, end);
            System.out.println(ans);
        }
    }
}

// } Driver Code Ends


class Solution {
    public int maxMeetings(int n, int start[], int end[]) {
        int a[][]=new int[n][2];
        for(int i=0;i<n;i++){
            a[i][0]=start[i];
            a[i][1]=end[i];
        }
        Arrays.sort(a,(o1,o2)->o1[1]-o2[1]);
        
        int ans=1, time=a[0][1];
        
        for(int i=1;i<n;i++){
            if(a[i][0]>time){
                ans++;
                time=a[i][1];
            }
        }
        return ans;
    }
}

File: Difficulty: Easy/N meetings in one room/n-meetings-in-one-room.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends

class Solution {
  public:
    // Function to find the maximum number of meetings that can
    // be performed in a meeting room.
    static bool cmp(pair<int,int>a,pair<int,int>b){
      return b.second > a.second;
  }
    int maxMeetings(vector<int>& start, vector<int>& end) {
        // Your code here
        vector<pair<int,int>>v;
    int n=start.size();
    for(int i=0;i<n;i++){ v.push_back({start[i],end[i]});}
    sort(v.begin(),v.end(),cmp);
    int eend=v[0].second;
    int ct=1;
    for(int i=1;i<n;i++){
        if(v[i].first>eend){
            ct++;
            eend=v[i].second;
        }
    }
    return ct;
    }
};


//{ Driver Code Starts.

int main() {

    int t;
    cin >> t;
    cin.ignore();
    while (t--) {
        string input;
        int num;
        vector<int> start;

        getline(cin, input);
        stringstream s2(input);
        while (s2 >> num) {
            start.push_back(num);
        }

        vector<int> end;
        getline(cin, input);
        stringstream s22(input);
        while (s22 >> num) {
            end.push_back(num);
        }

        Solution ob;
        int ans = ob.maxMeetings(start, end);
        cout << ans << endl;
        cout << "~" << endl;
    }
    return 0;
}
// } Driver Code Ends